### PR TITLE
feat: gitpodify project

### LINKED ISSUE #189: [FEATURE] Gitpodify Project
### ⭐ Suggestion

Make the project ready for gitpod

### 💻 Use Cases

_No response_

### ❌ Related Problems

_No response_

### PR DESCRIPTION
**Pull request by Cahllagerfeld**
closes #189 

some small issues to resolve:

- [x] await docker-compose before startup
- [x] discuss preview
- [x] adjust readme
- [x] add gitpod-app to project or eddiehub

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -65,11 +63,10 @@ jobs:
         uses: actions/setup-node@v2
         with:
           node-version: ${{ matrix.node-version }}
-      - name: Authenticate with GitHub package registry
-        run: echo "//npm.pkg.github.com/:_authToken=${{ secrets.GITHUB_TOKEN }}" > ~/.npmrc
       - name: install dependencies
         run: npm ci
         env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (eddiejaoude) [inline .github/workflows/build.yml:69]: Oh nice! this is much cleaner 👍

File: .npmrc [hunk 2]
```diff
@@ -1 +1,2 @@
 @cahllagerfeld:registry=https://npm.pkg.github.com
+//npm.pkg.github.com/:_authToken=${GH_TOKEN}
```
Comment (Cahllagerfeld) [inline .npmrc:2]: In Gitpod envrionment variables can be set per project.
The idea is, that the contributor needs to have one `GH_TOKEN` in his env.

I need to test this locally if that works how I want to

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -50,9 +47,10 @@ jobs:
         with:
           name: artifacts
           path: dist
-      - name: npm install
+      - name: install dependencies
         run: npm ci
         env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (Cahllagerfeld) [inline .github/workflows/docker.yml:53]: adjusted all actions to use the `GH_TOKEN` env-variable

File: .github/workflows/docker.yml [hunk 4]
```diff
@@ -83,6 +81,8 @@ jobs:
         with:
           context: .
           push: true
+          secrets: |
```
Comment (Cahllagerfeld) [inline .github/workflows/docker.yml:84]: Also needed to adjust the dockerfile & docker-action. 
I'm not sure here if it works as expected we probably have to test it first.

**Review by Cahllagerfeld — COMMENTED** [review]
Just a few comments

File: .github/workflows/docker.yml (hunk 3 above)
Comment (eddiejaoude) [reply .github/workflows/docker.yml:53]: Nice 🔥

File: .gitpod.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# List the start up tasks. Learn more https://www.gitpod.io/docs/config-start-tasks/
+tasks:
+  - name: Install npm depdencies
+    init: |
+      npm i -g npm
+      npm install
+    command: |
+      npm i -g npm
+      docker-compose up
```
Comment (Cahllagerfeld) [inline .gitpod.yml:9]: As Pauline said until theres a workaround the user needs to start it manually once the docker spin up finished.

File: .npmrc (hunk 2 above)
Comment (pawlean) [reply .npmrc:2]: Yeah, this makes sense. Nice one!

File: .gitpod.yml (hunk 5 above)
Comment (eddiejaoude) [reply .gitpod.yml:9]: ok, cool, we can add it to the docs in bold.

File: .gitpod.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# List the start up tasks. Learn more https://www.gitpod.io/docs/config-start-tasks/
+tasks:
+  - name: Install npm depdencies
+    init: |
+      npm i -g npm
+      npm install
+    command: |
+      npm i -g npm
+      sleep 30
```
Comment (Cahllagerfeld) [inline .gitpod.yml:9]: Added sleep here
`gp await-port` doesnt seem to work with stargate, as stargate opens the ports before the actual image is initialized successfull. (At least I have the impression)

Comment (eddiejaoude) [reply .gitpod.yml:9]: oh ok. If we need the first version to have a sleep I think that is ok, we can have an issue/todo for it, where we can improve it for the next version.

**Comment by Cahllagerfeld** [comment]
I think this can be checked for merge now (:

**Review by eddiejaoude — APPROVED** [review]
Awesome work 💥

**Pull request merged**